grpc_client.cpp

```diff
@@ -392,6 +532,138 @@ Datum ConvertValueListToArray(const ValueList& value_list)
     return PointerGetDatum(result_array);
 }
 
+void ValueToJsonbValue2(const Value& value, JsonbParseState **pstate, JsonbValue** r)
```

Comment:
ValueToJsonb had to be rewritten to support recursive values while propagating the "json state". It was tested with recursive values.